### DISCUSSION TITLE
Can't Use Ome with HLS (or LLHLS) stream

### DISCUSSION THREAD

**Discussion by antonymarion** [OvenMediaLabs/OvenMediaEngine#1662] [Help]
**Describe the bug**
I start Ome with Docker then I stream an hls stream using ffmpeg then I can't load it with OvenMediaPlayer

**To Reproduce**
1. `docker run -d \
-p 1935:1935 \
-p 3333:3333 \
-p 3334:3334 \
-p 3478:3478 \
-p 9000:9000 \
-p 9999:9999/udp \
-p 4000:4000/udp \
--name ovenmediaengine \
airensoft/ovenmediaengine:latest
`
2. `ffmpeg -v verbose -stream_loop -1 -re -i ./filename.m3u8  -c:v copy -c:a aac -ar 44100 -ab 128k -ac 2 -strict -2 -flags +global_header -bsf:a aac_adtstoasc -bufsize 3000k -f hls https://your_domain_or_localhost.net:1935/app/llhls.m3u8` 
3. Launch OvenMediaplayer over https://your_domain_or_localhost.net with url hls 

**Expected behavior**
It should show the hls stream

**Logs**
On Portainer I can see the following logs over OvenMediaEngine, but nothing else 
![image](https://github.com/user-attachments/assets/160c14b9-1e13-4334-ace3-967522f1ebcc)

On ffmpeg in verbose mode, I do not see the frames sent to ovenMediaEngine whereas on a test server (https://ffmpeg-hls-test.datastream.tebi.io/test.m3u8) with the same ffmpeg command I can see them
![image](https://github.com/user-attachments/assets/5dff4609-df37-4c07-9359-75f9e049bc06)

**Server (please complete the following information):**
 - OS: Windows
 - OvenMediaEngine Version: latest
 - Branch: main

**Player (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Browser Chrome

**Additional context**
Using the same ome server (deployed with docker or locally with docker) with rtmp and webRTC everything works fine

**Comment by naanlizard** [comment]
Afaik ome doesn't support his ingest, especially not like you're trying to
do. Use rtmp and make sure your ports are forwarded properly

**Reply by antonymarion** [reply]
Not sure to understand, you mean that OME cannot take an HLS stream as an input?

**Comment by naanlizard** [comment]
Maybe it can, but I'm not familiar with how it works if it does

Try rtmp

**Reply by antonymarion** [reply]
With rtmp, as I said I have no issue; but I want to test now LLHLS to have low latency (around seconds) and use DVR mainly.

As I do not know any DVR capability with WebRTC and OME with its player I want to try OME with LLHLS

**Reply by naanlizard** [reply]
Yes, stream rtmp to ome, and then use your llhls player. Ome does the conversion

**Reply by antonymarion** [reply]
OK so that is my issue then.

Let's say I am working locally on localhost using Docker.

Here is my command for streaming, it works well:

`ffmpeg -d -stream_loop -1 -re -i myVideo.mp4 -c copy -f flv -y rtmp://localhostt:1935/app/llhls`

Then what is the expected playback url in ovenplayer?
![image](https://github.com/user-attachments/assets/cfe9a125-f3e1-4471-96e8-612fee330293)

Because the one above does not work for me

**Comment by naanlizard** [comment]
You have a typo in your ffmpeg command. You might get better feedback for
testing using a gui program like obs, it would immediately tell you if you
can connect or not. I'm not sure ffmpeg does the same

Aside from the typo, your playback url is incorrect. Check the documentation

https://airensoft.gitbook.io/ovenmediaengine/streaming/low-latency-hls#:~:text=if%20you%20use%20the%20default%20configuration%2C%20you%20can%20start%20streaming%20with%20the%20following%20url%3A

You are maybe confusing yourself naming your stream as llhls. I might
recommend changing it to something like "test123" or something. You'll also
get more useful information from the logs

**Reply by antonymarion** [reply]
Oooh thks, great feedback, did not understand the naming convention from OME => always end by llhsl.m3u8 with key before the slash

**Reply by antonymarion** [reply]
By the way I did not change my ffmpeg command so no typo I believe ;) Just not read the OME doc carefully my bad!
